### ISSUE TITLE
Implement haskell-send-line, haskell-send-region, haskell-send-paragraph be wanted

### ISSUE THREAD

**Issue by abelew** [haskell/haskell-mode#892]
Greetings,
  I am new to haskell, but comfortable with emacs.  My google searches suggested that it is against the philosophy of haskell-mode to send individual lines/regions/paragraphs to an inferior-haskell.  Since I don't yet know enough haskell to understand why that is a good thing, I wrote myself those functions so that I may type '2+2' in my buffer and hit C-c C-n to evaluate it (for example).
  I haven't yet figured out how to get it to reprint the lambda> and so the output of each evaluation appears next to the existing lamda prompt, but that doesn't bother me nearly as much as not being able to evaluate arbitrary code did.
  If there is any interest, I will happily send along a merge request or diff.
  Thank you for your time.

**Comment by gracjan** [comment]
Hi @abelew,

Hands with keyboard and Emacs Lisp knowledge are surely welcome.

In case you are just looking for a topic to hack on we have some prepared:

https://github.com/haskell/haskell-mode/issues?q=is%3Aopen+is%3Aissue+label%3Awell-defined-task

If you have some functionality already implemented then we are open to pull requests. Note that we will be able to accept the code if it is general enough so that many users of haskell-mode find it useful. If in doubt fire a pull request and we can start from there.

**Comment by abelew** [comment]
-----BEGIN PGP SIGNED MESSAGE-----
Hash: SHA256

  Greetings,
  I implemented those functions (poorly).  For my money they would be
 usable by other people, but only because I used them to figure out
simple haskell concepts without having to write -> save -> :l in the
lambda> buffer.

  I certainly understand if the existing functions are not
useful/wanted.  If you wish to decide for yourself, they may be found at:
  https://github.com/abelew/99problems/blob/master/haskell/haskell.el

  If you wish, I can easily make them a little less stupid (replace
the setq with let bindings and maybe get the haskell buffer to print
the incoming input rather than just evaluate it) and submit a pull
request.
Conversely, you are welcome to take the defuns if they are useful.

  I checked out the set of topics, and I could likely help once I
understand haskell properly -- but for now I drew a pretty solid blank
on all of them except using cc-mode preprocessor highlighting -- I can
likely get that to work with little or no foreknowledge.

  I hope you have a good day,
 -trey

On 09/23/2015 10:30 AM, Gracjan Polak wrote:
> Hi @abelew,
> 
> Hands with keyboard and Emacs Lisp knowledge are surely welcome.
> 
> In case you are just looking for a topic to hack on we have some
> prepared:
> 
> https://github.com/haskell/haskell-mode/issues?q=is%3Aopen+is%3Aissue+label%3Awell-defined-task
>
>  If you have some functionality already implemented then we are
> open to pull requests. Note that we will be able to accept the code
> if it is general enough so that many users of haskell-mode find it
> useful. If in doubt fire a pull request and we can start from
> there.
> 
> 
> 
> --- Reply to this email directly or view it on GitHub: 
> https://github.com/haskell/haskell-mode/issues/892#issuecomment-142619372
>
-----BEGIN PGP SIGNATURE-----
Version: GnuPG v2

iF4EAREIAAYFAlYCz0gACgkQzgGyqJcfxqvSJAD+MWuFElvDBdRldRcbi70gne0m
2sulN0kyA+HoK0ljJkAA/RHe34OAoECuHrfR7BEmgn/RKNvGIJLWmf2T2V/Ykv6i
=k5cz
-----END PGP SIGNATURE-----
